doc-content/jbpm-docs/src/main/asciidoc/BPMN2/con_custom-tasks-overview.adoc

```diff
@@ -0,0 +1,195 @@
+[cols="2", options="header"]
+|===
+
+|Custom task type
+|Custom task node
+
+|Rest
+|image:BPMN2/bpmn-rest-custom-task.png[]
+
+|Email
+|image:BPMN2/bpmn-email-custom-task.png[]
+
+|Log
+|image:BPMN2/bpmn-log-custom-task.png[]
+
+|WebService
+|image:BPMN2/bpmn-webservice-custom-task.png[]
+
+|Milestone
+|image:BPMN2/bpmn-milestone.png[]
+
+|DecisionTask
+|image:BPMN2/bpmn-decision-task-custom.png[]
+
+|BusinessRuleTask
+|image:BPMN2/bpmn-business-rule-custom-task.png[]
+
+|KafkaPublishMessages
+|image:BPMN2/bpmn-kafkapublishmessages-task.png[]
+
+|===
+
+For more information about enabling or disabling custom tasks in {CENTRAL}, see <<manage-service-tasks-proc_custom-tasks>>.
+
+In the BPMN modeler, you can configure the following general properties for a selected custom task:
+
+.General custom task properties
+[cols="30%,70%", options="header"]
+|===
+|Label
+|Description
+
+|*Name*
+|Identifies the name of the task. You can also double-click the task node to edit the name.
+
+|*Documentation*
+|Describes the task. The text in this field is included in the process documentation, if applicable.
+
+|*Is Async*
+|Determines whether this task is invoked asynchronously.
+
+|*AdHoc Autostart*
+|Determines whether this is an ad hoc task that is started automatically. This option enables the task to automatically start when the process is created instead of being started by a signal event.
+
+|*On Entry Action*
+|Defines a Java, JavaScript, or MVEL script that directs an action at the start of the task.
+
+|*On Exit Action*
+|Defines a Java, JavaScript, or MVEL script that directs an action at the end of the task.
+
+|*SLA Due Date*
+|Specifies the date when the service level agreement (SLA) expires.
+
+|*Assignments*
+|Defines data input and output for the task.
+
+|===
+
+.Rest
+A rest custom task is used to invoke a remote RESTful service or perform an HTTP request from a process.
+
+image::BPMN2/bpmn-rest-custom-task.png[]
+
+To use the rest custom task, you can set the URL, HTTP method, and credentials in the process modeler. When a process reaches a rest custom task, it generates an HTTP request and returns the response as a string.
+
+You can click *Assignments* in the *Properties* panel to open the *REST Data I/O* window and add data input and output as required. For example, enter the following data inputs in *Data Inputs and Assignments* fields which are required to execute a rest custom task:
+
+* *ConnectTimeout*: Connection timeout (in seconds). The default value is 60 seconds.
+* *ContentData*: Data you want to send. This attribute is mandatory for `POST` and `PUT` requests.
+* *Method*: Method of the endpoint called, such as `GET`, and `POST`.
+* *Password*: Password for Authentication.
+* *ReadTimeout*: Timeout (in seconds) on response. The default value is 60 seconds.
+* *Url*: Endpoint URL for the REST service.
+* *Username*: User name for Authentication.
+* *Headers*: Headers to pass for REST call, such as `content-type=text/html`.
```

Comment:
```suggestion
* *Url*: Endpoint URL for the REST service. Mandatory property.
* *Method*: Method of the endpoint called, such as `GET`, and `POST`. Default value is `GET`.
* *ContentType*: Data type in case of sending data. Mandatory property for `POST`, `PUT`.
* *ContentTypeCharset*:
* *Content*: 
* *ContentData*: Data you want to send. This attribute is mandatory for `POST` and `PUT` requests.
* *ConnectTimeout*: Connection timeout (in seconds). The default value is 60 seconds.
* *ReadTimeout*: Timeout (in seconds) on response. The default value is 60 seconds.
* *Username*: User name for Authentication.
* *Password*: Password for Authentication.
* *AuthUrl*: Url that is handling Authentication.
* *AuthType*:
* *HandleResponseErrors*: Optional parameter that instructs handler to throw errors in case of non successful response codes (other than 2XX).
* *ResultClass*: Fully qualified class name of the class that response should be transformed to, if not given string format will be returned.
* *AcceptHeader*: Accept header value.
* *Headers*: Headers to pass for REST call, such as `content-type=text/html`.
* *AcceptCharset*:
```

Reply:
Ditto